src/Features/CSharp/Portable/ConvertToRawString/ConvertRegularStringToRawStringCodeRefactoringProvider.cs

```diff
@@ -128,20 +164,148 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             SyntaxFormattingOptions options,
             CancellationToken cancellationToken)
         {
-            return kind switch
+            var characters = CSharpVirtualCharService.Instance.TryConvertToVirtualChars(token);
+            Contract.ThrowIfTrue(characters.IsDefaultOrEmpty);
+
+            // If the user asked to remove whitespace then do so now.
+            if (kind == ConvertToRawKind.MultiLineWithoutLeadingWhitespace)
+                characters = CleanupWhitespace(characters);
+
+            return kind == ConvertToRawKind.SingleLine
+                ? ConvertToSingleLineRawString(token, characters)
+                : ConvertToMultiLineRawIndentedString(document, token, options, characters, cancellationToken);
+        }
+
+        private static VirtualCharSequence CleanupWhitespace(VirtualCharSequence characters)
+        {
+            using var _ = ArrayBuilder<VirtualCharSequence>.GetInstance(out var lines);
+
+            // First, determine all the lines in the content.
+            BreakIntoLines(characters, lines);
+
+            // Remove the leading and trailing line if they are all whitespace.
+            while (lines.Count > 0 && AllWhitespace(lines.First()))
+                lines.RemoveAt(0);
+
+            while (lines.Count > 0 && AllWhitespace(lines.Last()))
+                lines.RemoveAt(lines.Count - 1);
+
+            if (lines.Count == 0)
+                return VirtualCharSequence.Empty;
```

Comment:
Should this check be moved up and prevent the fix from being offered? I can't think of why someone would have a bunch of whitespace is a verbatim string, but offering to convert it to an empty string is probably not what they want to happen.

Reply:
fair point.  Updated to not offer in that case.